src/string.js

```diff
@@ -172,3 +172,22 @@ export function trimStart(str) {
 
   return (str + '_').trim().slice(0, -1);
 }
+
+/**
+ * Pads the beginning of a string with a substring to a target length.
+ * @param {string} s
+ * @param {number} targetLength
+ * @param {string} padString
+ * @return {*} TODO(#23582): Specify return type
+ */
+export function padStart(s, targetLength, padString) {
```

Comment:
I think this could use a couple of simple tests (in `test/unit/test-string.js`).